src/api/lib/statistics_calculations.rb

```diff
@@ -19,9 +21,12 @@ def self.packages(limit, timelimit, prj_filter, pkg_filter)
   private_class_method :packages
 
   def self.projects(limit, timelimit, prj_filter)
-    Project.where(updated_at: timelimit..Time.now).
-      where('name REGEXP ?', prj_filter).
-      order('updated_at DESC').
+    projects = Project.all
+    projects = projects.where(updated_at: timelimit..Time.now) if timelimit
+    if prj_filter
+      projects = projects.where('name REGEXP ?', prj_filter)
+    end
+    projects.order('updated_at DESC').
       limit(limit).
       pluck(:name, :updated_at).
       map { |name, at| [at, name, :project] }
```

Comment:
Nitpicking, and nothing you touched anyway... but this could be a ``map!`.

Reply:
hmm, it returns that value - so I think map is perfectly fine

Reply:
both do: https://ruby-doc.org/core-2.5.0/Array.html#method-i-map-21

Reply:
ah, so it's not creating a new array

Reply:
This is a much bigger story -> https://github.com/openSUSE/open-build-service/pull/6429 

(and I mentioned you in the commit log as revenche :)